src/BSON/DBPointer.c

```diff
@@ -68,6 +68,31 @@ static bool php_phongo_dbpointer_init_from_hash(php_phongo_dbpointer_t* intern,
 	return false;
 } /* }}} */
 
+HashTable* php_phongo_dbpointer_get_properties_hash(phongo_compat_object_handler_type* object, bool is_temp) /* {{{ */
+{
+	php_phongo_dbpointer_t* intern;
+	HashTable*              props;
+
+	intern = Z_OBJ_DBPOINTER(PHONGO_COMPAT_GET_OBJ(object));
+
+	PHONGO_GET_PROPERTY_HASH_INIT_PROPS(is_temp, intern, props, 2);
+
+	if (!intern->ref) {
+		return props;
+	}
+
+	{
+		zval ref, id;
+
+		ZVAL_STRING(&ref, intern->ref);
+		ZVAL_STRING(&id, intern->id);
+		zend_hash_str_update(props, "ref", sizeof("ref") - 1, &ref);
+		zend_hash_str_update(props, "id", sizeof("id") - 1, &id);
```

Comment:
No need to change anything here, but getting to the second class in the code review I thought "I suppose this is now 20 times the same change", and I immediately catch the different code organisation between this and the Binary class up there. No need to change since I'd like to refactor this code in general: macros that call the combination of `ZVAL_*` and `zend_hash_str_update` would help since they could handle the scoping (if it's necessary, see earlier comment).

Reply:
Created [PHPC-1962](https://jira.mongodb.org/browse/PHPC-1962) to track this.